File: 2022/221231-dojo/java/app/src/main/java/Solution.java
class Solution {
    public String solution(String s, int n) {
        String accumulate = "";
        String upperCase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        String lowerCase = "abcdefghijklmnopqrstuvwxyz";
        String[] words = s.split("");

        for (int i = 0; i < words.length; i += 1) {
            if (!upperCase.contains(words[i]) && !lowerCase.contains(words[i])) {
                accumulate += " ";
                continue;
            }

            if (upperCase.contains(words[i])) {
                int index = upperCase.indexOf(words[i]);

                if (index + n > 25) {
                    accumulate += upperCase.charAt(index + n - 26);
                    continue;
                }

                if (index + n <= 25) {
                    accumulate += upperCase.charAt(index + n);
                }
            }

            if (!upperCase.contains(words[i])) {
                int index = lowerCase.indexOf(words[i]);

                if (index + n > 25) {
                    accumulate += lowerCase.charAt(index + n - 26);
                    continue;
                }

                if (index + n <= 25) {
                    accumulate += lowerCase.charAt(index + n);
                }
            }
        }

        return accumulate;
    }
}

File: 2022/221231-dojo/javascript/solution.js
export default function solution(s, n) {
  const upperCaseArray = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');
  const lowerCaseArray = upperCaseArray.map((value) => value.toLowerCase());

  const words = s.split('');

  return words.map((value) => {
    if (upperCaseArray.find((element) => element === value)) {
      const index = upperCaseArray.findIndex((item) => item === value);

      if (index + n <= 25) {
        return upperCaseArray[index + n];
      }

      return upperCaseArray[index + n - 26];
    }

    if (lowerCaseArray.find((element) => element === value)) {
      const index = lowerCaseArray.findIndex((item) => item === value);

      if (index + n <= 25) {
        return lowerCaseArray[index + n];
      }

      return lowerCaseArray[index + n - 26];
    }

    return ' ';
  }).reduce((pre, cur) => pre + cur, '');
}
